ign_rviz_common/src/rviz/common/frame_manager.cpp

```diff
@@ -35,24 +35,17 @@ namespace common
  * Creates a tf subscription and binds callback to it.
  */
 FrameManager::FrameManager(rclcpp::Node::SharedPtr _node)
-: QObject(), frameCount(0)
+: QObject(), node(std::move(_node))
 {
-  this->node = std::move(_node);
-
   tfBuffer = std::make_shared<tf2_ros::Buffer>(this->node->get_clock());
   tfListener = std::make_shared<tf2_ros::TransformListener>(*tfBuffer);
-
-  this->subscriber = this->node->create_subscription<tf2_msgs::msg::TFMessage>(
-    "/tf", 10,
-    std::bind(&FrameManager::tf_callback, this, std::placeholders::_1));
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 void FrameManager::setFixedFrame(const std::string & _fixedFrame)
 {
   std::lock_guard<std::mutex>(this->tf_mutex_);
```

Comment:
Do you still need the mutex? I don't see it being used anywhere else. The only resource that is being shared across the methods is fixedFrame.

Reply:
Yeah, it since setting the fixed frame is not an atomic operation it could lead to errors